### DISCUSSION TITLE
New built-in packages resolve feature just dropped for Unity

### DISCUSSION THREAD

**Discussion by ComradeVanti** [openupm/openupm-cli#336] [General]
Seems like Unity just introduced some sort of new built-in package resolving logic in 2022.3.29.

![image](https://github.com/openupm/openupm-cli/assets/31240807/f5d78c41-4484-46b2-97df-8eeb0f79d55b)

Not quite sure how this works, but maybe can be useful to us. Just wanted to put it out there.

https://unity.com/releases/editor/whats-new/2022.3.29

**Comment by favoyang** [comment]
I'm not sure what that means.

I have tried to update one project from Editor 2022.3.17f1 to 2022.3.29f1. And I received the below message once.

```
Packages were changed.
Update Mode: updateDependencies

The following packages were updated:
  com.unity.collab-proxy from version 2.2.0 to 2.3.1
  com.unity.visualscripting from version 1.9.1 to 1.9.4
```

This is nothing new, when you upgrade to another Editor version, Unity will update some preferred packages for you once, as part of the "version migration" process. These two are not built-in packages either, they are still hosted on the Unity registry.

Then I try to install an old version of a built-in package `com.unity.render-pipelines.universal`

```
"com.unity.render-pipelines.universal": "14.0.9",
```

Unity actually installs `com.unity.render-pipelines.universal@14.0.11` for me, no matter which version I specify in the `manifest.json`. Because Unity 2022.3.29f1 shipped with the built-in package `com.unity.render-pipelines.universal@14.0.11`.

**Reply by favoyang** [reply]
I think they might have added an extra layer to find updated built-in packages, which are hosted somewhere else instead of the Unity Registry. For example, even when `com.unity.render-pipelines.universal@14.0.12` is released, users can still access it without upgrading to a newer Unity Editor. But I can not prove it.

This seems to conflict with the definition of "built-in." As I understand it, built-in packages are meant to be tied to a specific Unity Editor version to prevent update compatibility issues.

**Reply by ComradeVanti** [reply]
I will look into this soon, maybe also analyzing the web-requests Unity sends to its servers so we can get a better understanding of it. This is relevant for example for #155
